src/Features/LanguageServer/Protocol/Features/Options/WorkspaceConfigurationOptionsStorage.cs

```diff
@@ -13,9 +12,14 @@ internal static class WorkspaceConfigurationOptionsStorage
     public static WorkspaceConfigurationOptions GetWorkspaceConfigurationOptions(this IGlobalOptionService globalOptions)
         => new(
             CacheStorage: globalOptions.GetOption(Database),
-            EnableOpeningSourceGeneratedFiles: globalOptions.GetOption(EnableOpeningSourceGeneratedFilesInWorkspace) ??
-                                               globalOptions.GetOption(EnableOpeningSourceGeneratedFilesInWorkspaceFeatureFlag),
+            EnableOpeningSourceGeneratedFiles:
+                globalOptions.GetOption(EnableOpeningSourceGeneratedFilesInWorkspace) ??
+                globalOptions.GetOption(EnableOpeningSourceGeneratedFilesInWorkspaceFeatureFlag),
             DisableRecoverableText: globalOptions.GetOption(DisableRecoverableText),
+            SourceGeneratorExecution:
+                globalOptions.GetOption(SourceGeneratorExecution) ??
+                (globalOptions.GetOption(SourceGeneratorExecutionBalancedFeatureFlag) ? SourceGeneratorExecutionPreference.Balanced : SourceGeneratorExecutionPreference.Automatic),
```

Comment:
by default we'll be in automatic mode.  but this approach allows us to try out A/B experiments like checking what impcat this has on internal users if we set to balanced.